### DISCUSSION TITLE
web version creates folder inside "All vaults" instead of "My vault" per default

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#3298] [Q&A]
### Subject of the issue

web version creates folder inside "All vaults" instead of "My vault" per default

### Deployment environment

I am using https://bitwarden.garudalinux.org as an end user and have no access to the admin page

### Steps to reproduce

I have no organizations, so "My Vault" is the only vault

1. open vaultwarden from the web
2. "All Vaults" is selected by default, if not select it
3. press plus button to add a folder
4. name it, eg "test from web"
5. select "My Vault"

### Expected behaviour

folder "test from web" must be listed in "My Vault" folders
 
### Actual behaviour

folder "test from web" is not listed in "My Vault" folders, but only in "All Vaults".

Firefox extensions and Android version creates the new folders correctly inside "My Vault" and they are also displayed in "All Vaults"

**Comment by BlackDex** [comment]
Seems like a client issue to me.
First, test this with the `testing` tagged images of Vaultwarden.
Second, test this on vault.bitwarden.com if it also happens there it's definitely a client issue. Also, since you say ot works on the extensions, it probably isn't a server side bug.

If this also happens on the Bitwarden Cloud, then i suggest to report this at the bitwarden/clients repo.

**Comment by ghost** [comment]
>  First, test this with the testing tagged images of Vaultwarden

sorry, the instance is not mine, I am just an end user

> test this on vault.bitwarden.com

does not have this issue, in the main page both instances states 2023.2.0

**Comment by BlackDex** [comment]
Thanks for the report. We have to investigate.

**Comment by stefan0xC** [comment]
I'm not sure what the issue is. Does the folder disappear when you click my vault? Testing it with the latest `testing` image, i.e. `337cbfaf` seems to work <s>just fine</s>.

[Peek 2023-03-02 22-14.webm](https://user-images.githubusercontent.com/509385/222555395-bc39b5ee-128a-4766-b56e-4764c24bb581.webm)

**Reply by stefan0xC** [reply]
Okay, so the folders don't disappear even though they are empty. <s>The issue is that they SHOULD disappear. So the observed behavior is the opposite of what was intended because</s> funnily enough what does disappear are the folders with items in it. See below for the linked issue.

update: after reading the pull request it seems like I was wrong again. :sweat_smile:

**Reply by stefan0xC** [reply]
Okay, according to the [pull request](
https://github.com/bitwarden/clients/pull/4700) that fixes the issue the folders should never disappear on My Vault

> * "All vaults" and "My vault" should always show all folders
> * Organization vault should only show non-empty folders and the "No folder" folder
> * Ensures the "Folders" filter section is always visible

**Comment by ghost** [comment]
# My Vault's folders
![My Vaults](https://user-images.githubusercontent.com/15078341/222557440-959ef9f3-1c73-4227-a40d-07508193d2b4.png)

# Vault's folders
![Vault](https://user-images.githubusercontent.com/15078341/222557446-dfeb0d41-fd28-453d-82fb-160c60f51478.png)

**Comment by BlackDex** [comment]
The issue is that the folder is visible everywhere, no matter which org you select. At least that is what i just noticed on my instance.

I can't debug right now my self, so no clue why this is.

**Comment by stefan0xC** [comment]
According to the `/api/config` endpoint the server you are using is still on version 024d12db

@BlackDex fixed some v2023.2.0 API calls in 7ec00d3850f0c058314b4bc4a3c5d1094882c434 so this might be why? (Have not tested it.)

**Comment by BlackDex** [comment]
> According to the `/api/config` endpoint the server you are using is still on version [024d12d](https://github.com/dani-garcia/vaultwarden/commit/024d12db08820d89eb7fd8551a5773dc9f295feb)
> 
> @BlackDex fixed some v2023.2.0 API calls in [7ec00d3](https://github.com/dani-garcia/vaultwarden/commit/7ec00d3850f0c058314b4bc4a3c5d1094882c434) so this might be why? (Have not tested it.)

I tested it on my personal hosted instance. It should be updated, but, i haven't checked to be honest haha.

**Comment by stefan0xC** [comment]
> The issue is that the folder is visible everywhere, no matter which org you select. At least that is what i just noticed on my instance.
> 

Okay, tested it with two folders that have items in it and on my instance "My Vault" always shows all folders except for the two folders that have items in it. Seems to be tracked upstream here https://github.com/bitwarden/clients/issues/4805 (oh, and it's already fixed so in the next release of the web-vault this should not happen anymore.)

**Comment by BlackDex** [comment]
> > The issue is that the folder is visible everywhere, no matter which org you select. At least that is what i just noticed on my instance.
> 
> Okay, tested it with two folders that have items in it and on my instance "My Vault" always shows all folders except for the two folders that have items in it. Seems to be tracked upstream here [bitwarden/clients#4805](https://github.com/bitwarden/clients/issues/4805) (oh, and it's already fixed so in the next release of the web-vault this should not happen anymore.)

Yea, i thought it was expected behavior indeed. Because, you can assign any cipher, not just personal items to a folder.
